staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```
@@ -462,6 +458,37 @@ func (cfgCtlr *configController) digestConfigObjects(newPLs []*flowcontrol.Prior
 	return suggestedDelay, utilerrors.NewAggregate(errs)
 }
 
+func updateOrApply(client flowcontrolclient.FlowSchemaInterface, fsu fsStatusUpdate, asFieldManager string) error {
+	if utilfeature.DefaultFeatureGate.Enabled(features.ServerSideApply) {
+		return apply(client, fsu, asFieldManager)
+	}
+
+	return update(client, fsu, asFieldManager)
+}
+
+func apply(client flowcontrolclient.FlowSchemaInterface, fsu fsStatusUpdate, asFieldManager string) error {
+	applyOptions := metav1.ApplyOptions{FieldManager: asFieldManager, Force: true}
+
+	// the condition field in fsStatusUpdate holds the new condition we want to update.
+	// TODO: this will break when we have multiple conditions for a flowschema
+	_, err := client.ApplyStatus(context.TODO(), toFlowSchemaApplyConfiguration(fsu), applyOptions)
+	return err
+}
+
+func update(client flowcontrolclient.FlowSchemaInterface, fsu fsStatusUpdate, asFieldManager string) error {
+	// the condition field in fsStatusUpdate holds the new condition we want to update.
+	// TODO: this will break when we have multiple conditions for a flowschema
+	patchBytes, err := makeFlowSchemaConditionPatch(fsu.condition)
+	if err != nil {
+		// should never happen because these conditions are created here and well formed
+		panic(fmt.Sprintf("Failed to json.Marshall(%#+v): %s", fsu.condition, err.Error()))
```

not directly related to this PR, maybe we can return an error here